db/compaction/compaction_picker.cc

```diff
@@ -744,13 +744,9 @@ Compaction* CompactionPicker::CompactRange(
 
       input_level_total += input_file_size;
 
-      if (input_level_total + output_level_total >= limit) {
+      if (input_level_total + output_level_total > limit) {
         covering_the_whole_range = false;
-        // still include the current file, so the compaction could be larger
-        // than max_compaction_bytes, which is also to make sure the compaction
-        // can make progress even `max_compaction_bytes` is small (e.g. smaller
-        // than an SST file).
-        inputs.files.resize(i + 1);
```

Comment:
nit: could you replace this comment so it is clear that `resize(i)` means to leave out `inputs[i]`? maybe also clarify that we want to ensure `output_level_total <= limit`